doc/boot_protocol.rst

```diff
@@ -0,0 +1,274 @@
+Security Kernel Loader
+######################
+
+The Security Kernel Loader (SKL) is an implementation of a Secure Loader for
+the SKINIT CPU instruction. The SKL loader takes a minimal approach relying on
+the DRTM Preamble, e.g. the system bootloader, to load into memory and
+configure the Security Kernel that the SKL will measure and execute.
+
+Memory Layout
+*************
+
+The DRTM Preamble is responsible for allocating a contiguous block of physical
+memory and will ensure it is marked as reserved in the memory map. The
+necessary size for the allocation will be found in the SKL header. The DRTM
+Preamble will place the SKL at the beginning of the allocation.
+
+Below is the layout of how the memory allocation is segmented::
+
+      +---     Security Kernel Loader    ---+
+      |                                     |
+      v                                     v
+      +--------+----------+-------------+-------------+--------------+
+      | Header |   Code   |  Boot Tags  | IOMMU Table |  Log Buffer  |
+      +--------+----------+-------------+-------------+--------------+
+      ^                                 ^                            ^
+      |                                 |                            |
+      +                                 +                            +
+  sklbuf_base                          60kb                   header.skl_size
```

Comment:
"how the memory is laid out", but do we really want to include this?  It is liable to get stale quite quickly.

Reply:
well, hopefully it won't go stale quickly. This is an attempt to get to a stable ABI.

Reply:
Ah,I see the confusion. The IOMMU Table region I am proposing is not a static/compile time generated region. This layout was specified specifically with the assumption that the IOMMU tables would be generated at run time. During runtime the SLK would locate the end of the boot tags and then take the first 4k aligned address after, provided it was not outside the 64k SKINIT protected region. Thus why i was trying to put the restriction on the boot protocol that boot tags could not exceed past the 60k offset. What needs to be determined at compile time is how much space in total does the SLK need for the SLB (64k) + additional external space, and communicate that to the bootloader via the header. Now we could rename header.skl_size to header.alloc_req or something similar but what is important is that this protocol explains how the boot loader is to use the information. As for the diagram, I think it should stay but with detail removed to reduce confusion and not mix internal details with information needed by external entities.